### DISCUSSION TITLE
Does LSP4IJ support multiple instances of Rust language servers in the same project?

### DISCUSSION THREAD

**Discussion by l2dy** [redhat-developer/lsp4ij#494] [Q&A] [answered]
For example, in an [Aya](https://github.com/aya-rs/aya) project, the main program can be analyzed with any recent version of Rust, but the eBPF counterpart must use nightly versions, as specified in [its rust-toolchain.toml](https://github.com/aya-rs/aya-template/blob/main/%7B%7Bproject-name%7D%7D-ebpf/rust-toolchain.toml).

In this case, there is a conflict between `rust-toolchain.toml` files or the lack thereof. Is it possible for LSP4IJ to support multiple workspaces with different instances of language servers in the same IntelliJ project, or would I have to ask users to use separate projects (if multiple projects opened in different windows do not interfere with each other) or pick a best-effort version with some heuristics?

**Comment by angelozerr** [comment, marked as answer]
No it is not possible today, the language server is a singleton for a given project.

**Reply by l2dy** [reply]
FYI, the proper solution seems to be https://blog.jetbrains.com/idea/2024/09/intellij-idea-2024-3-eap/#workspaces-in-intellij-idea, and LSP4IJ doesn't need to change its model.
